cmd/otk-gen-partition-table/main.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+
+	"github.com/osbuild/images/internal/buildconfig"
+	"github.com/osbuild/images/internal/cmdutil"
+	"github.com/osbuild/images/internal/common"
+	"github.com/osbuild/images/internal/otkdisk"
+	"github.com/osbuild/images/pkg/blueprint"
+	"github.com/osbuild/images/pkg/disk"
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+type Input struct {
+	Properties InputProperties   `json:"properties"`
+	Partitions []*InputPartition `json:"partitions"`
+
+	Modifications InputModifications `json:"modifications"`
+}
+
+type InputProperties struct {
+	UEFI        InputUEFI `json:"uefi"`
+	BIOS        bool      `json:"bios"`
+	Type        string    `json:"type"`
```

Comment:
Can we make this an enum?

Reply:
I considered it but since this is an input data structure I think making it an enum string doesn't give us much benefit.  But we should validate the values somehow so perhaps an enum makes that more explicit.